### ISSUE TITLE
[Bug]: Permission Errors When Managing Files with Shopify CLI on a New macOS User Account

### ISSUE THREAD

**Issue by jclarkdev** [Shopify/cli#3083]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

Files created within the Shopify project directory should inherit the parent directory's permissions, allowing the creating user to edit them without requiring elevated permissions (sudo).

### Actual behavior

Files created in the Shopify project directory require elevated permissions (sudo) to edit, even when they are created by the current user within their home directory. This leads to a disruptive workflow where each save operation needs 'Retry as Sudo'. I can only assume this is being caused by the way shopify-cli handles the files in a theme, but I'm not able to bridge the gap in my mind of why this would be the case.

### Verbose output

```shell
~/ecomdept/clients/jason-wu git:(press-page)
-> shopify theme dev --store jason-wu-studio --verbose

2023-11-09T18:08:15.714Z: Running command theme dev
  shopify:theme:dev init version: @oclif/core@2.11.7 argv: [ '--store', 'jason-wu-studio', '--verbose' ] +0ms
2023-11-09T18:08:15.779Z: Ensuring that the user is authenticated with the Theme API with the following scopes:
[]

2023-11-09T18:08:15.780Z: Ensuring that the user is authenticated with the Admin API with the following scopes for the store jason-wu-studio.myshopify.com:
[]

2023-11-09T18:08:15.782Z: Getting session store...
2023-11-09T18:08:15.787Z: Validating existing session against the scopes:
[
  "openid",
  "https://api.shopify.com/auth/shop.admin.graphql",
  "https://api.shopify.com/auth/shop.admin.themes",
  "https://api.shopify.com/auth/partners.collaborator-relationships.readonly",
  "https://api.shopify.com/auth/shop.storefront-renderer.devtools",
  "https://api.shopify.com/auth/partners.app.cli.access",
  "https://api.shopify.com/auth/destinations.readonly"
]
For applications:
{
  "adminApi": {
    "scopes": [],
    "storeFqdn": "jason-wu-studio.myshopify.com"
  }
}

2023-11-09T18:08:15.788Z: Sending Identity Introspection request to URL: https://accounts.shopify.com/oauth/introspection
2023-11-09T18:08:15.789Z: Sending POST request to URL https://accounts.shopify.com/oauth/introspection
With request headers:
 - User-Agent: Shopify CLI; v=3.50.0
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-11-09T18:08:17.950Z: Request to https://accounts.shopify.com/oauth/introspection completed in 2137 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"73262d27b49ec50ac98337414f1ed26b"
 - x-request-id: 06ca0fa2-da36-44d0-b89b-4fe230ac5f80
    
2023-11-09T18:08:17.957Z: The identity token is valid: true
2023-11-09T18:08:17.957Z: 
The validation of the token for application/identity completed with the following results:
- It's expired: false
- It's invalid in identity: false
  
2023-11-09T18:08:17.958Z: The current session is valid but needs refresh. Refreshing...
2023-11-09T18:08:17.959Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=refresh_token&access_token=atkn_CjQI15-0qgYQ99e0qgZSJggBEhABuIHoGbRCh6uGyGEvIvmHGhBc13AIpCBJoocoQAdXubQiEkDAt3cAf3ieRQNc-wa4vwQcZEhWObhuKdQqBMmcNp5zi9_JPDW5FgPA56JyM59xPhFICbTx-7yOkYXNhpM1AO8B&refresh_token=atkn_CiEI15-0qgYQ17nSqwaiARIKEFzXcAikIEmihyhAB1e5tCISQDVP9rgQxvWRxIjJAYCE-st4u1PuEobdPkiin7I5-uIdxqz8LQlBEo2tqyLQ1J5XwGTny6rygF71p8XMgBLfEAw&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3
With request headers:
 - User-Agent: Shopify CLI; v=3.50.0
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-11-09T18:08:18.219Z: Request to https://accounts.shopify.com/oauth/token?grant_type=refresh_token&access_token=atkn_CjQI15-0qgYQ99e0qgZSJggBEhABuIHoGbRCh6uGyGEvIvmHGhBc13AIpCBJoocoQAdXubQiEkDAt3cAf3ieRQNc-wa4vwQcZEhWObhuKdQqBMmcNp5zi9_JPDW5FgPA56JyM59xPhFICbTx-7yOkYXNhpM1AO8B&refresh_token=atkn_CiEI15-0qgYQ17nSqwaiARIKEFzXcAikIEmihyhAB1e5tCISQDVP9rgQxvWRxIjJAYCE-st4u1PuEobdPkiin7I5-uIdxqz8LQlBEo2tqyLQ1J5XwGTny6rygF71p8XMgBLfEAw&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3 completed in 258 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"d8952732fb617e7543581a85328b4750"
 - x-request-id: add7e7e9-46f0-44c0-8610-f3b4859ad271
    
2023-11-09T18:08:18.223Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=271e16d403dfa18082ffb3d197bd2b5f4479c3fc32736d69296829cbb28d41a6&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.app.cli.access&subject_token=****
With request headers:
 - User-Agent: Shopify CLI; v=3.50.0
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-11-09T18:08:18.223Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=ee139b3d-5861-4d45-b387-1bc3ada7811c&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.storefront-renderer.devtools&subject_token=****
With request headers:
 - User-Agent: Shopify CLI; v=3.50.0
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-11-09T18:08:18.223Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=32ff8ee5-82b8-4d93-9f8a-c6997cefb7dc&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fdestinations.readonly&subject_token=****
With request headers:
 - User-Agent: Shopify CLI; v=3.50.0
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-11-09T18:08:18.223Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=7ee65a63608843c577db8b23c4d7316ea0a01bd2f7594f8a9c06ea668c1b775c&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.graphql+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.themes+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.collaborator-relationships.readonly&subject_token=****&destination=https%3A%2F%2Fjason-wu-studio.myshopify.com%2Fadmin
With request headers:
 - User-Agent: Shopify CLI; v=3.50.0
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-11-09T18:08:18.457Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=7ee65a63608843c577db8b23c4d7316ea0a01bd2f7594f8a9c06ea668c1b775c&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.graphql+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.themes+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.collaborator-relationships.readonly&subject_token=****&destination=https%3A%2F%2Fjason-wu-studio.myshopify.com%2Fadmin completed in 228 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"4aac0d6982d1d7165d3d8a5090b23a40"
 - x-request-id: 70e20543-abe0-4330-9c48-96a276077f8b
    
2023-11-09T18:08:18.461Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=ee139b3d-5861-4d45-b387-1bc3ada7811c&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.storefront-renderer.devtools&subject_token=**** completed in 234 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"8358394e395e7a8bea47e3929144a7e6"
 - x-request-id: 85435884-c985-48ac-9c38-c74d2f363a11
    
2023-11-09T18:08:18.467Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=32ff8ee5-82b8-4d93-9f8a-c6997cefb7dc&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fdestinations.readonly&subject_token=**** completed in 240 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"b19f722ffc99b0a186644c575c587643"
 - x-request-id: 9ac00cc8-fb93-4047-8a10-e0c5ca66ce9e
    
2023-11-09T18:08:18.475Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=271e16d403dfa18082ffb3d197bd2b5f4479c3fc32736d69296829cbb28d41a6&scope=https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.app.cli.access&subject_token=**** completed in 250 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"ad147d0890dd859021026c313a5f8467"
 - x-request-id: 98963229-b564-4c27-9376-19519abaa37f
    
2023-11-09T18:08:18.477Z: Setting session store...
2023-11-09T18:08:18.502Z: Ensuring that the user is authenticated with the Storefront API with the following scopes:
[]

2023-11-09T18:08:18.502Z: Getting session store...
2023-11-09T18:08:18.503Z: Validating existing session against the scopes:
[
  "openid",
  "https://api.shopify.com/auth/shop.admin.graphql",
  "https://api.shopify.com/auth/shop.admin.themes",
  "https://api.shopify.com/auth/partners.collaborator-relationships.readonly",
  "https://api.shopify.com/auth/shop.storefront-renderer.devtools",
  "https://api.shopify.com/auth/partners.app.cli.access",
  "https://api.shopify.com/auth/destinations.readonly"
]
For applications:
{
  "storefrontRendererApi": {
    "scopes": []
  }
}

2023-11-09T18:08:18.503Z: Sending Identity Introspection request to URL: https://accounts.shopify.com/oauth/introspection
2023-11-09T18:08:18.503Z: Sending POST request to URL https://accounts.shopify.com/oauth/introspection
With request headers:
 - User-Agent: Shopify CLI; v=3.50.0
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-11-09T18:08:18.699Z: Request to https://accounts.shopify.com/oauth/introspection completed in 194 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"5cff1b452fab7740c5586af666493d72"
 - x-request-id: b353788f-1fcd-41cd-bcc2-e30e040daf78
    
2023-11-09T18:08:18.700Z: The identity token is valid: true
2023-11-09T18:08:18.700Z: 
The validation of the token for application/identity completed with the following results:
- It's expired: false
- It's invalid in identity: false
  
2023-11-09T18:08:18.713Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/ruby -v
  · Working directory: /Users/ecomdept/ecomdept/clients/jason-wu

2023-11-09T18:08:18.732Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle -v
  · Working directory: /Users/ecomdept/ecomdept/clients/jason-wu

2023-11-09T18:08:19.051Z: Reading the content of file at /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/Gemfile...
2023-11-09T18:08:19.054Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle config set --local path /Users/ecomdept/Library/Caches/shopify-gems-nodejs
  · Working directory: /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-11-09T18:08:19.248Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle config set --local without development:test
  · Working directory: /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-11-09T18:08:19.431Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle install
  · Working directory: /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-11-09T18:08:19.905Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle exec /usr/local/opt/ruby/bin/ruby /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify theme token --admin atkn_Cp8DCJLAtKoGELL4tKoGYpADCAESEEw7JYgsvUoVm6TXXpaTwXEaqAFodHRwczovL2FwaS5zaG9waWZ5LmNvbS9hdXRoL3BhcnRuZXJzLmNvbGxhYm9yYXRvci1yZWxhdGlvbnNoaXBzLnJlYWRvbmx5IGh0dHBzOi8vYXBpLnNob3BpZnkuY29tL2F1dGgvc2hvcC5hZG1pbi5ncmFwaHFsIGh0dHBzOi8vYXBpLnNob3BpZnkuY29tL2F1dGgvc2hvcC5hZG1pbi50aGVtZXMgDSinUzIQwQ1LpJvLRa6kQJ6Cp8UJhTokNDlhMTkzYzctZGUzOS00YzY1LWIzOGYtMGY3NDNkM2VmMzdkQgdBY2NvdW50ShAHRGPfqslC24obO-5BxH2fUlN7InN1YiI6ImZiZGIyNjQ5LWUzMjctNDkwNy04ZjY3LTkwOGQyNGNmZDdlMyIsImlzcyI6Imh0dHBzOi8vYWNjb3VudHMuc2hvcGlmeS5jb20ifWIQLRYZGFo1SpKLyLNq54GYI2oQcPUU7rQ_R4iSpgCFG17zVBJAgYoBFnjQ6kXGTnn_MedgGiX1bJPCk-sYasNFOIzUVQH0RydoEmRPh6ZM_SyOwlqLRb1Gzao1u_FaTpsxe1w1DA --sfr atkn_CqMCCJLAtKoGELL4tKoGYpQCCAESEMDFHwgzsUkynCL-cnt4PC0aPmh0dHBzOi8vYXBpLnNob3BpZnkuY29tL2F1dGgvc2hvcC5zdG9yZWZyb250LXJlbmRlcmVyLmRldnRvb2xzINhSKKdTOiQ0OWExOTNjNy1kZTM5LTRjNjUtYjM4Zi0wZjc0M2QzZWYzN2RCB0FjY291bnRKEAdEY9-qyULbihs77kHEfZ9SU3sic3ViIjoiZmJkYjI2NDktZTMyNy00OTA3LThmNjctOTA4ZDI0Y2ZkN2UzIiwiaXNzIjoiaHR0cHM6Ly9hY2NvdW50cy5zaG9waWZ5LmNvbSJ9YhAtFhkYWjVKkovIs2rngZgjahBw9RTutD9HiJKmAIUbXvNUEkB_C57COFkbye2vyQdemd2XHMm_iAc9oxdjH9ywE7xrmuntLqU06a5uQARYko6N4cPx98dN19ism38LBinP7JQG
  · Working directory: /Users/ecomdept/ecomdept/clients/jason-wu

/usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/pstore.rb:633:in `initialize': Permission denied @ rb_sysopen - /Users/ecomdept/.cache/shopify/.db.pstore (Errno::EACCES)
        from /usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/pstore.rb:633:in `new'
        from /usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/pstore.rb:633:in `open_and_lock_file'
        from /usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/pstore.rb:569:in `transaction'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/lib/shopify_cli/db.rb:61:in `set'
        from /usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/forwardable.rb:240:in `set'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/lib/shopify_cli/migrator.rb:24:in `store_last_migration_date'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/lib/shopify_cli/migrator.rb:20:in `migrate'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/lib/shopify_cli.rb:180:in `<module:ShopifyCLI>'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/lib/shopify_cli.rb:42:in `<top (required)>'
        from <internal:/usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from <internal:/usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify:9:in `require'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/load_shopify.rb:18:in `<top (required)>'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify:45:in `require_relative'
        from /usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify:45:in `<main>'
Error
    at execCLI2 (file:///usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/dist/public/node/ruby.js:64:15)
    at async refreshTokens (file:///usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/dist/cli/services/dev.js:113:9)
    at async Dev.run (file:///usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/dist/cli/commands/theme/dev.js:18:51)
    at async Dev._run (/usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/theme/node_modules/@oclif/core/lib/command.js:117:22)
    at async Config.runCommand (/usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/cli/node_modules/@oclif/core/lib/config/config.js:329:25)
    at async run (/usr/local/Cellar/shopify-cli/3.50.0/libexec/lib/node_modules/@shopify/cli/node_modules/@oclif/core/lib/main.js:89:16)
```

### Reproduction steps

1. Log into a non-admin macOS user account
2. Verify Homebrew and Shopify CLI using sudo as recommended for a multi-user environment
3. Clone into a shopify theme using `git clone` or `shopify theme pull`
4. Attempt to create a new file with a standard command (touch filename.liquid, or create the file in the VSCode GUI).
5. Encounter permission error: Error: EACCES: permission denied.
6. Use sudo touch filename.liquid to successfully create the file.
7. Edit the file in a text editor and attempt to save.
8. Encounter permission prompt to 'Retry as Sudo' for each save operation.

In the end, `sudo touch` and "Save > Retry As Sudo..." succeeds, but this is jumping through hoops for what should be a simple workflow.

### Operating System

Intel Macbook Pro, MacOS Ventura 13.4.1

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.50.0

### Shell

iTerm2, zsh, VSCode integrated terminal

### Node version (run `node -v` if you're not sure)

v21.1.0

### What language and version are you using in your application?

ruby 2.6.10p210

**Comment by jclarkdev** [comment]
**Update:** I was able `git clone` the project, `touch sections/main-press.liquid`, and then edit that new file and save it without any trouble.

I think the discrepancy is that I ran `sudo shopify theme pull` last time (since I cant use shopify-cli without running sudo from this new user account), and thus the permissions for the newly pulled theme were under the root user and required me to run `sudo` when writing to them.

I'm tempted to close this because technically I've resolved my personal roadblock, but I still think there is a little bit too much friction when operating shopify-cli from another user. Therefore I'm going to leave this open in hopes for a better resolution, or more detailed instructions on how I should be operating shopify-cli in a multi-user environment.

Maybe I need to install the shopify-cli binary independently on this new user instead of relying on homebrew?
